src/value_generators/include/value_generators/FrequencyMap.hpp

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <exception>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+namespace genny {
+
+/**
+ * Genny frequency map. A list of pairs <string, count>
+ *
+ * The list is unsorted.
+ */
+class FrequencyMap {
```

Comment:
I _think_ this is only in a separate header because of unit-testing, ya? Downside is that this can now be used outside of DocumentGenerator.cpp even though it's really an internal implementation detail that I think we'd otherwise just put in the .cpp directly. Is there a better way to indicate that this is "private"?